modules/creating-manifest-file-multiple-interfaces-br-ex.adoc

```diff
@@ -0,0 +1,177 @@
+// Module included in the following assemblies:
+//
+// * support/troubleshooting/troubleshooting-network-issues.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="creating-manifest-file-multiple-interfaces-br-ex_{context}"]
+= Creating an additional OVN-Kubernetes `br-ex` bridge network
+
+You can create an additional `br-ex` bridge network that has a different configuration to the default `br-ex` bridge. You must define the additional bridge in the target interface definition of a machine config manifest file.  
+
+After you create and edit the manifest file, the Machine Config Operator completes tasks in the following order:
+
+* Drains nodes in singular order based on the selected machine config pool.
+* Injects Ignition configuration files into each node, so that each node receives the additional `br-ex` bridge network configuration.
+* Restarts the node with the modified override file.
+* Executes the `configure-ovs.sh` shell script that references the new interface definition.
+* Adds `br-ex` and `br-ex1` to the host node.
+* Uncordons the nodes. After all the nodes return to the `Ready` state, the Operator applies the `k8s.ovn.org/l3-gateway-config` annotation to each node.
+
+For more information about useful situations for the additional `br-ex` bridge and a situation that always requires the default `br-ex` bridge, see "Configuration for a localnet topology".
+
+.Procedure
+
+. Convert the rule to Base64 encoded by running the following command:
++
+[source,terminal]
+----
+$ base64 /path/to/file/
+----
+
+. Complete one of the following steps to define a secondary interface for your network. Ensure that the secondary interface is not bound to the `br-ex` bridge.
++
+[IMPORTANT]
+====
+Do not use the Kubernetes NMState Operator to define or a `NodeNetworkConfigurationPolicy` (NNCP) manifest file to define the secondary interface.
+====
++
+.. Define a secondary interface in a machine config manifest file:
++
+[source,yaml]
```

Comment:
our machine-config definition should now include all of the interfaces at once (since we're defining eno1, eno2 and bond1, we need to reflect in the same config):

This machine-config can replace the one at the bottom where you indicate 12-${ROLE}-sec-bridge-cni. They are the same, so you can also move up the instruction to define the role here above this block -->

~~~

. Prepare the environment variables. Replace `<machine_role>` with the node role, such as `worker`, and replace `<interface_name>` with the name of your additional `br-ex` bridge name.
+
[source,terminal]
----
$ export ROLE=<machine_role>
----
+
[source,terminal]
----
$ export INTERFACE=<interface_name>
----
~~~

machine-config definition:

~~~
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: worker
  name: 12-${ROLE}-sec-bridge-cni
spec:
  config:
    ignition:
      version: 3.2.0
    storage:
      files:
      - contents:
          source: data:;base64,<base-64-encoded-contents-for-bond1.conf>
        path: /etc/NetworkManager/system-connections/bond1.nmconnection
        filesystem: root
        mode: 0600
      - contents:
          source: data:;base64,<base-64-encoded-contents-for-eno1.conf>
        path: /etc/NetworkManager/system-connections/eno1.nmconnection
        filesystem: root
        mode: 0600
      - contents:
          source: data:;base64,<base-64-encoded-contents-for-eno2.conf>
        path: /etc/NetworkManager/system-connections/eno2.nmconnection
        filesystem: root
        mode: 0600
~~~

Reply:
:heavy_check_mark: